modules/lvm2/udiskslvm2daemonutil.c

```diff
@@ -308,11 +309,32 @@ variant_reader_watch_child (GPid     pid,
 static void
 variant_reader_destroy (gpointer user_data)
 {
+  char err_buf[64];
+  int rc = 0;
   struct VariantReaderData *data = user_data;
-
+  gint fd = g_io_channel_unix_get_fd(data->output_channel);
   g_source_remove (data->output_watch);
   g_io_channel_unref (data->output_channel);
   g_free (data);
+
+  /* Make sure to close the underlying file descriptor, this apparently is
+   * not done when the channel gets cleaned up.  Documentation doesn't mention
+   * it, but this keeps us from exhausting all the file descriptors we have for
+   * the process.
+   */
+  rc = close(fd);
+  if (rc != 0 )
+    {
+      int ec = errno;
+      if (0 == strerror_r(ec, err_buf, sizeof(err_buf)))
```

Comment:
Wouldn't g_strerror( ) be better here? It should be thread-safe too and save us from using our own string buffer.